### PR TITLE
Use Guzzle to for version checking

### PR DESCRIPTION
**Pull request by damcou**
Two changes in the ExtensionNotification class :
- Use Guzzle to perform the call to Github API
- Log the exception if there's any

### CODE REVIEW

**Comment by peterjaap** [comment]
LGM!

**Comment by vmalyk** [comment]
Hi!

I have couple concerns about this PR:
1) By default Guzzle isnt`t present in  Magento Composer requirements, But might be to add in composer.json
2) Using Guzzle for 1 request - it looks overhead for me and might be better way -  use Magento Framework classes for HTTP client factory ([here](https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/HTTP/ClientFactory.phpl) and used [CURL class](https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/HTTP/Client/Curl.php) . In this case you won`t add extra requirement from Point 1.

@damcou , @bsuravech ,
Please review,
Thanks.

**Comment by peterjaap** [comment]
@vmalyk 

Yes it is:

```
composer why guzzlehttp/guzzle
magento/framework                  102.0.3  requires  guzzlehttp/guzzle (^6.3.3)  
magento/product-community-edition  2.3.3    requires  guzzlehttp/guzzle (^6.3.3) 
```

**Comment by vmalyk** [comment]
@peterjaap ,

but it`s not used for magento 2.2.x.

**Comment by peterjaap** [comment]
Yes it's introduced in 2.3.3. So we could just add this as a requirement for this extension for people who run out-dated M2 versions, and then it'll be installed automatically.

**Comment by peterjaap** [comment]
But I'm not set on having Guzzle at all. It's just a nice library to use for this stuff and it's included anyway for the newest M2 version so I'm all in favor of using it.

**Comment by damcou** [comment]
Hello @peterjaap and @vmalyk , 

Thanks again for your help on this topic.
I think that adding guzzle as a composer requirement shouldn't be a problem and I'd stick to this solution if it's confirmed that it solves the https://github.com/algolia/algoliasearch-magento-2/issues/941 issue.

**Comment by vmalyk** [comment]
@damcou , please check latest commit - looks as now we have duplicated "require" section. Can you fix and add guzzle to top section?

Thnx!

**Comment by damcou** [comment]
@vmalyk Thanks again for spotting that :) . It's fixed.

File: Model/ExtensionNotification.php [hunk 1]
```diff
@@ -128,12 +138,19 @@ private function checkExtensionVersion()
...
+                    'headers' => self::APPLICATION_JSON_HEADER,
+                ]
             );
-            $this->repoData = json_decode($json);
+
+            if ($response->getStatusCode() != 200) {
+                throw new \Exception($response->getReasonPhrase());
+            }
+
+            $this->repoData = json_decode($response->getBody()->getContents());
```
Comment (Serializator) [inline Model/ExtensionNotification.php]: Magento has its own interface for (de)serialization (`Magento\Framework\Serialize\SerializerInterface`) with a specific implementation for JSON (de)serialization (`Magento\Framework\Serialize\Serializer\Json`)

Comment (bsuravech) [reply Model/ExtensionNotification.php]: Love that this suggestion is coming from the Serializator.

Comment (damcou) [reply Model/ExtensionNotification.php]: Thanks a lot for your suggestion @Serializator , I made the requested change ;)

**Pull request merged**